docs/components/connectors/out-of-the-box-connectors/azure-open-ai.md

```diff
@@ -0,0 +1,164 @@
+---
+
+The **Azure OpenAI Connector** is an outbound Connector that allows you to interact with
+[Azure OpenAI](https://azure.microsoft.com/en-us/products/ai-services/openai-service) models from your BPMN processes.
+
+The **Azure OpenAI Connector** currently supports only prompt operations:
+[`completions`](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#completions),
+[`chat completions`](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions), and
+[`completions extensions`](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#completions-extensions).
+
+Refer the [official models documentation](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models)
+to find out whether a desired model supports mentioned operations.
+
+## Prerequisites
+
+To begin using the **Azure OpenAI Connector**, you need to have an Azure OpenAI resource created, and deployed.
+A valid Azure OpenAI API key is also required.
+
+Learn more at the [official Azure OpenAI portal entry](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/create-resource).
+
+## Create an Azure OpenAI Connector task
+
+To use the **Azure OpenAI Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon, or create a new Connector task using the **Append Connector** context menu. Follow our [guide on using Connectors](/components/connectors/use-connectors/index.md) to learn more.
+
+## Make your Azure OpenAI Connector executable
+
+To work with the **Azure OpenAI Connector**, fill all mandatory fields.
+
+## Authentication
+
+Fill the **API key** field with a valid Azure OpenAI API key.
+[Learn more](https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=command-line%2Cpython-new&pivots=rest-api#retrieve-key-and-endpoint) how to obtain a key.
+
+### Create a new Connector secret
+
+We advise you to keep your **API key** safe and avoid exposing it in the BPMN `xml` file by creating a secret:
+
+1. Follow our [guide for creating secrets](/components/console/manage-clusters/manage-secrets.md).
+2. Name your secret (i.e `AZURE_OAI_SECRET`) so you can reference it later in the Connector.
+
+## Operation
+
+Select desired operation from the **Operation** dropdown.
+Fill the **Resource name**, the **Deployment ID**, and the **API version** related to your operation. Make sure that the
+deployed [model](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models) supports selected operation.
+
+### Completion
+
+Refer to the [official documentation page](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#completions)
+to learn more about correct usage, limitations, examples, and versions.
+
+### Chat completion
+
+Refer to the [official documentation page](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions)
```

Comment:
Would it be better just to add an admonition saying "For completion details, refer to the Microsoft reference documentation."?

Reply:
Totally makes sense. Updated, please take a look if that looks better?

Reply:
I condensed a little -- WDYT?